app/models/grda_warehouse/ad_hoc_batch.rb

```diff
@@ -58,18 +58,28 @@ def process!
     if check_header!
       match_clients!
     else
-      self.import_errors = "CSV headers do not match expected headers: #{self.class.csv_headers.join(',')}; found: #{csv.headers.join(',')}"
+      self.import_errors = "CSV headers do not match expected headers: #{self.class.csv_headers.join(',')}; found: #{headers_from_csv.join(',')}"
     end
     self.completed_at = Time.current
     save(validate: false)
   end
 
   private def csv
-    @csv ||= CSV.parse(content, headers: true)
+    @csv ||= if content_type.in?(['text/plain', 'text/csv'])
```

Comment:
A completely empty file gets content type `application/x-empty` which the Excel parser can't recognize (it is expects a Zip, and the zip processor can't deal with empty files).